src/components/icon/icon.js

```diff
@@ -20,25 +21,55 @@ angular.module('material.components.icon', [
  * @restrict E
  *
  * @description
- * The `<md-icon>` directive is an element useful for SVG icons
+ * The `<md-icon>` directive is an element useful for showing an icon
  *
  * @usage
  * <hljs lang="html">
- *  <md-icon icon="/img/icons/ic_access_time_24px.svg">
- *  </md-icon>
+ *  <md-icon md-icon-key="icon-android" md-icon-size=""></md-icon>
  * </hljs>
  *
  */
-function mdIconDirective() {
+function mdIconDirective($mdIcon) {
   return {
+    scope: {
+      mdIconKey: '@',
+      mdIconSize: '@',
+      mdIconColor: '@'
+    },
     restrict: 'E',
-    template: '<object class="md-icon"></object>',
-    compile: function(element, attr) {
-      var object = angular.element(element[0].children[0]);
-      if(angular.isDefined(attr.icon)) {
-        object.attr('data', attr.icon);
+    replace: true,
+    template: '<span class="icon"></span>',
```

Comment:
The [Polymer core-icon](https://www.polymer-project.org/docs/elements/core-elements.html#core-icon) handles it by allowing the developer to provide an `alt` attribute, copying that content to `aria-label` and setting `role="img"`. If alt is provided as an empty string, it is hidden from the accessibility layer. If neither of those are present, it copies the key to the `aria-label`. This sounds like it could work well for Angular Material, too.